charts/vineyard/templates/worker/statefulset.yaml

```diff
@@ -91,6 +91,9 @@ spec:
             - --spill_upper_rate
             - "{{ include "vineyard.spill.upperRate" . }}"
             {{- end }}
+            - --metrics
```

Comment:
Does it mean that vineyard reports metrics via exporter? But how vineyardd manage to find out which is the allocated exporter port?

Reply:
> Does it mean that vineyard reports metrics via exporter?

By vineyardd itself. The `--metrics` field means vineyardd will print internal metrics info (only about vineyard object) to the stdout and the `log_dir` field means vineyard will store these metrics info under a directory.

> But how vineyardd manage to find out which is the allocated exporter port?

vineyardd doesn't need to know the port, we have an external exporter to convert these logs (/var/log/vineyard) to metrics via https://github.com/fluid-cloudnative/fluid/pull/3649/files#diff-6109bb7b99b63ccc2d90ea9d7c1ca8dece4d3873aabfc77280196643e7f6593cR162.

Reply:
Thanks! Got it now. One more question is that, does Vineyardd automatically rotate its log messages?

Reply:
> does Vineyardd automatically rotate its log messages?

No, vineyardd will preserve all log messages.

>And what will happen if the external exporter fail over?

It will convert all logs to metrics again.

>Will the external exporter recover all the metrics from the all the logs?

Yes, from the log file.